### PR TITLE
refactor(output_format): support print key value list style for text format

### PR DESCRIPTION
**Pull request by aviatco**
# 📥 Pull Request

### Overview
This PR introduces a new output formatting option that displays data in a clean, human-readable key-value format instead of the default JSON structure. This enhancement improves the user experience by providing a more intuitive way to view command outputs.

### Key Changes
1. Core Infrastructure Updates
[FabricCLIOutput](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/src/fabric_cli/core/fab_output.py:80): Added show_key_value_pretty parameter to control the new formatting option
[print_output_format()](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/src/fabric_cli/utils/fab_ui.py:98): Enhanced the main output function to support the new formatting parameter
2. New Key-Value Pretty Formatting Function
[_print_entries_key_value_pretty_style()](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/src/fabric_cli/utils/fab_ui.py:495): Implemented the core function that converts data into key-value format
[_format_key_to_pretty_name()](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/src/fabric_cli/utils/fab_ui.py:525): Added utility to transform snake_case and camelCase keys into readable "Title Case" format
3. Smart Output Selection Logic
[Text Output Handler](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/src/fabric_cli/utils/fab_ui.py:360): Updated _print_output_format_result_text() to conditionally use key-value formatting when show_key_value_pretty=True
Format Priority: JSON format takes precedence over key-value formatting to maintain backward compatibility
4. Comprehensive Test Coverage
[Unit Tests](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/tests/test_utils/test_fab_ui.py:563-732): Added extensive test coverage including:
Key-value pretty format activation and output verification
JSON format precedence testing
Input validation and error handling
Key formatting transformation tests
### Features
Output Format Examples
#### Before (JSON):

{
  "logged_in": "true",
  "account_name": "johndoe@example.com"
}

#### After (Key-Value Pretty):

Logged In: true
Account Name: johndoe@example.com

### Key Formatting Intelligence
Converts snake_case → Title Case (e.g., logged_in → Logged In)
Converts camelCase → Spaced Title Case (e.g., accountName → Account Name)
Handles both single dictionaries and lists of dictionaries
Adds spacing between multiple entries for clarity
Integration Points
Works with existing [print_output_format()](vscode-webview://140b27luhbsrq1hbc0vdgdjtt9i64jjl41prlmlq500v6fhgp4ce/src/fabric_cli/utils/fab_ui.py:92) calls throughout the codebase
Maintains full backward compatibility
Respects output format precedence (JSON > key-value > default)
Technical Implementation
Type Safety: Proper input validation with meaningful error messages
Performance: Minimal overhead, only processes data when explicitly enabled
Extensibility: Clean separation of concerns allows easy future enhancements
Error Handling: Graceful fallback for invalid input types

### Testing
✅ Unit tests for all new functions and edge cases
✅ Integration tests with existing output pipeline
✅ Error handling and input validation coverage
✅ Format precedence verification

This enhancement provides users with a more readable alternative to JSON output while maintaining full compatibility with existing workflows and scripts.

**## NOTE The flag only affects text output mode; JSON output ignores this setting.**

### CODE REVIEW

File: src/fabric_cli/core/fab_output.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def __init__(
         error_code: Optional[str] = None,
         data: Optional[Any] = None,
         hidden_data: Optional[Any] = None,
+        show_key_value_pretty: bool = False,
```
Comment (ayeshurun) [inline src/fabric_cli/core/fab_output.py]: can we name it show_key_value_list or show_as_key_value_list or something similar (without the pretty :))

File: src/fabric_cli/utils/fab_ui.py [hunk 2]
```diff
@@ -486,3 +490,49 @@ def _get_visual_length(string: str) -> int:
...
+    """
+    if isinstance(entries, dict):
+        _entries = [entries]
+    elif isinstance(entries, list):
+        if not entries:
+            return
+        _entries = entries
+    else:
+        raise FabricCLIError(
+            ErrorMessages.Labels.invalid_entries_format(),
```
Comment (ayeshurun) [inline src/fabric_cli/utils/fab_ui.py]: the location of the error message should not be under Labels

File: src/fabric_cli/utils/fab_ui.py [hunk 3]
```diff
@@ -486,3 +490,49 @@ def _get_visual_length(string: str) -> int:
...
+            fab_constant.ERROR_INVALID_ENTRIES_FORMAT,
+        )
+
+    for entry in _entries:
+        for key, value in entry.items():
+            pretty_key = _format_key_to_pretty_name(key)
```
Comment (ayeshurun) [inline src/fabric_cli/utils/fab_ui.py]: convert_to_pascal_case

or

convert_to_title_case

File: src/fabric_cli/utils/fab_ui.py [hunk 4]
```diff
@@ -486,3 +490,49 @@ def _get_visual_length(string: str) -> int:
...
+    """Convert a snake_case or camelCase key to a Title Case pretty name.
+    
+    Args:
+        key: The key to format (e.g. 'logged_in' or 'accountName')
+        
+    Returns:
+        str: Formatted pretty name (e.g. 'Logged In' or 'Account Name')
+    """
+    # Replace underscores and camelCase with spaces
+    pretty = key.replace('_', ' ')
```
Comment (ayeshurun) [inline src/fabric_cli/utils/fab_ui.py]: this will print 'power_bi' as 'Power Bi' with small `i`.

File: src/fabric_cli/utils/fab_ui.py [hunk 5]
```diff
@@ -510,29 +511,41 @@ def _print_entries_key_value_pretty_style(entries: Any) -> None:
...
+        key: The key to format (e.g. 'user_id' or 'accountName')
         
     Returns:
-        str: Formatted pretty name (e.g. 'Logged In' or 'Account Name')
+        str: Formatted to title case name (e.g. 'User Id' or 'Account Name')
     """
     # Replace underscores and camelCase with spaces
     pretty = key.replace('_', ' ')
-    pretty = ''.join(' ' + char if char.isupper() else char for char in pretty).strip()
+    # pretty = ''.join(' ' + char if char.isupper() else char for char in pretty).strip()
```
Comment (ayeshurun) [inline src/fabric_cli/utils/fab_ui.py]: please add comment what this regex does

**Comment by ayeshurun** [comment]
please add changie entry

**Review by ayeshurun — APPROVED** [review]

File: src/fabric_cli/utils/fab_ui.py [hunk 6]
```diff
@@ -486,3 +491,62 @@ def _get_visual_length(string: str) -> int:
...
+    Args:
+        key: The key to format (e.g. 'user_id' or 'accountName')
+        
+    Returns:
+        str: Formatted to title case name (e.g. 'User Id' or 'Account Name')
+    """
+    # Replace underscores and camelCase with spaces
+    pretty = key.replace('_', ' ')
+    # Replacing the camelCase with spaces only if the previous character is not a space
+    pretty = re.sub(r'(?<!^)(?<! )(?=[A-Z])', ' ', pretty)
```
Comment (ohadedry) [inline src/fabric_cli/utils/fab_ui.py]: Is it possible that key will be something like UserID? in this case the output will be User I D.

Comment (ohadedry) [reply src/fabric_cli/utils/fab_ui.py]: I mean, what if the key contains multiple consecutive uppercase letters, is it possible scenario?

Comment (aviatco) [reply src/fabric_cli/utils/fab_ui.py]: updated the logic - now allowed only snake_case

File: src/fabric_cli/utils/fab_ui.py [hunk 7]
```diff
@@ -486,3 +491,83 @@ def _get_visual_length(string: str) -> int:
...
+        key: The key to format in snake_case format (e.g. 'user_id', 'account_name')
+        
+    Returns:
+        str: Formatted to title case name (e.g. 'User ID', 'Account Name')
+        
+    Raises:
+        ValueError: If the key is not in the expected underscore-separated format
+    """
+    # Allow letters, numbers, and underscores only
+    if not key.replace('_', '').replace(' ', '').isalnum():
```
Comment (ohadedry) [inline src/fabric_cli/utils/fab_ui.py:537]: Spaces are allowed? `.replace(' ', '')`

Comment (aviatco) [reply src/fabric_cli/utils/fab_ui.py:537]: no - not allowed - only snake case (with _)

**Review by ayeshurun — DISMISSED** [review]

File: src/fabric_cli/utils/fab_ui.py [hunk 8]
```diff
@@ -4,6 +4,7 @@
 import builtins
 import html
 import json
+import re
```
Comment (ohadedry) [inline src/fabric_cli/utils/fab_ui.py]: I think it is unused

**Review by ohadedry — APPROVED** [review]
